pkg/bootstrapper/hostname.go

```diff
@@ -0,0 +1,34 @@
+package bootstrapper
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+)
+
+func getAWSMetadataHostname() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return "", fmt.Errorf("unable to load config: %w\n", err)
+	}
+
+	client := imds.NewFromConfig(cfg)
+
+	hostnameRes, err := client.GetMetadata(context.TODO(), &imds.GetMetadataInput{
+		Path: "hostname",
+	})
+	if err != nil {
+		return "", fmt.Errorf("unable to retrieve the hostname from the EC2 instance: %w\n", err)
+	}
+
+	defer hostnameRes.Content.Close()
+	hostname, err := io.ReadAll(hostnameRes.Content)
+	if err != nil {
+		return "", fmt.Errorf("cannot read hostname from the EC2 instance: %w\n", err)
+	}
```

Comment:
Would recommend not wrapping the errors here and removing the the new lines as well

Reply:
done